dcrwallet.go

```diff
@@ -289,7 +290,22 @@ func rpcClientConnectLoop(legacyRPCServer *legacyrpc.Server, loader *wallet.Load
 				legacyRPCServer.SetChainServer(chainClient)
 			}
 			if cfg.EnableTicketBuyer {
-				startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				pm, err = startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				if err != nil {
+					log.Errorf("Unable to start ticket buyer: %v", err)
+					go func() {
+						loader.SetWaitFn(nil)
+					}()
+					return
+				}
+				pm.Start()
+				go func() {
```

Comment:
still confused, I don't see how this code is acquiring the loader mutex in a way that prevents the OpenExistingWallet RPC to execute.